src/hooks/useProfiles.ts

```diff
@@ -3,24 +3,31 @@ import { UseMutationOptions, UseQueryOptions, useMutation, useQuery } from 'reac
 import { useNavigate } from 'react-router';
 import { useTranslation } from 'react-i18next';
 
-import { createProfile, deleteProfile, enterProfile, listProfiles, updateProfile } from '#src/stores/ProfileController';
 import { useProfileStore } from '#src/stores/ProfileStore';
 import type { CommonAccountResponse, ListProfilesResponse, ProfileDetailsPayload, ProfilePayload } from '#types/account';
 import { useAccountStore } from '#src/stores/AccountStore';
 import type { GenericFormErrors } from '#types/form';
 import type { ProfileFormSubmitError, ProfileFormValues } from '#src/containers/Profiles/types';
 import { logDev } from '#src/utils/common';
+import ProfileController from '#src/stores/ProfileController';
+import { getModule } from '#src/modules/container';
+import AccountController from '#src/stores/AccountController';
 
 export const useSelectProfile = () => {
   const navigate = useNavigate();
+  const isLoggedIn = useAccountStore((s) => !!s.user);
 
-  return useMutation((vars: { id: string; pin?: number; avatarUrl: string }) => enterProfile({ id: vars.id, pin: vars.pin }), {
+  const accountController = isLoggedIn ? getModule(AccountController) : undefined;
```

Comment:
Repeating the suggestion not to do conditional module resolution, but to put the conditions closer to where they are really being evaluated.